pkg/apis/samples/v1/types.go

```diff
@@ -424,9 +428,23 @@ func (s *Config) ClusterNeedsCreds() bool {
 		s.Spec.ManagementState == operatorv1.Unmanaged {
 		return false
 	}
-	if s.Status.Conditions == nil {
+	if s.Status.Conditions == nil || len(s.Status.Conditions) == 0 {
 		return true
 	}
+
+	// some timing paths can lead to only the  config valid condition existing,
+	// so explicitly check it the import creds condition is even there yet
+	foundImportCred := false
+	for _, rc := range s.Status.Conditions {
+		if rc.Type == ImportCredentialsExist {
+			foundImportCred = true
+			break
+		}
+	}
+	if !foundImportCred {
+		return true
```

Comment:
why doesn't this return value care what the samples registry is?

Reply:
addressed in new push